app/js/components/viewSystemGroups/viewSystemGroups.directive.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+var componentsModule = require('../');
+
+/**
+ * @ngInject
+ */
+function viewSystemGroupsCtrl($scope,
+                              $mdToast,
+                              $state,
+                              System,
+                              Group) {
+    let undoDeleteGroup = null;
+
+    $scope.groupLimit = 4;
+    $scope.showAll = false;
+
+    function loadSystemGroups(uuid) {
+        System.getSystemGroups(uuid).success(function (groups) {
+            $scope.groups = groups;
+        });
+    }
+
+    $scope.toggleShowAll = function () {
+        $scope.showAll = !$scope.showAll;
+    };
+
+    $scope.getUUID = function () {
+        if ($scope.system.machine_id) {
+            return $scope.system.machine_id; // for legacy
+        }
+
+        return $scope.system.system_id;
+    };
+
+    $scope.groupSelected = function (group) {
+        // parent.parent === systemModal.controller.js
+        $scope.$parent.$parent.close();
+        Group.setCurrent(group);
+        $state.go('app.inventory');
+    };
+
+    $scope.deleteSystemFromGroup = function (group) {
+        Group.removeSystem(group, $scope.system).success(function () {
+            undoDeleteGroup = group;
+            loadSystemGroups($scope.getUUID());
+        });
+    };
```

Comment:
doesn't refresh the underlying page after removing a system. Should broadcast an event that the underlying pages are listening for so that they can refresh their systems.

Reply:
Might want to wait until the modal is closed before broadcasting the event or at least the toaster message.